test/processors/JsonSchemaInputProcessor.spec.ts

```diff
@@ -123,4 +124,79 @@ describe('JsonSchemaInputProcessor', function() {
         });
     });
 
-});
\ No newline at end of file
+    describe('reflectSchemaName()', function() {
+        test('should work', async function() {
+            const schema = {
+                properties: {
+                    prop: {
+                        type: "string",
+                    },
+                    allOfCase: {
+                        allOf: [
+                            {
+                                type: "string",
+                            },
+                            {
+                                type: "string",
+                            },
+                        ],
+                    }, 
+                },
+                patternProperties: {
+                    patternProp: {
+                        type: "string",
+                    }
+                },
+                dependencies: {
+                    dep: {
+                        type: "string",
+                    },
+                },
+                definitions: {
+                    def: {
+                        type: "string",
+                    },
+                    oneOfCase: {
+                        oneOf: [
+                            {
+                                type: "string",
+                            },
+                            {
+                                type: "string",
+                            },
+                        ],
+                    }, 
+                },
+                anyOf: [
+                    {
+                        type: "string",
+                    },
+                    {
+                        type: "string",
+                    },
+                ]
+            }
+            const expected = JsonSchemaInputProcessor.reflectSchemaName(schema) as any;
+
+            // properties
+            expect(expected.properties.prop['x-modelgen-inferred-name']).toEqual('prop');
```

Comment:
this is gonna be a problem if the same property name is used in multiple places, I would not recommend to use the prop name as is but rather path to that property. For example in this case I would do `RootPropertiesProp`. 

Or would you rather handle this in the simplifier? Not sure that is better.. What would the id then be? `propX`?

Reply:
Nope, I also handle such a case. I will update test.